### ISSUE TITLE
Hide software keyboard

### ISSUE THREAD

**Issue by bootstraponline** [linkedin/test-butler#41]
This seems like something test-butler should take care of?

> adb shell settings put secure show_ime_with_hard_keyboard 0
>
> http://www.drewhannay.com/2016/04/how-to-hide-android-soft-keyboard-in.html

**Comment by drewhannay** [comment]
Seems reasonable. I think we also want it in our emulator startup script still, cause it's annoying for devs to have to change it manually.

**Comment by bootstraponline** [comment]
Is [UiAutomation.html#executeShellCommand](https://developer.android.com/reference/android/app/UiAutomation.html#executeShellCommand(java.lang.String)) the right approach for setting this via on device Java code? 

The issue with a startup script is it's not going to work on test clouds where the emulator isn't subject to user control.

**Comment by drewhannay** [comment]
Yeah, the startup script comment was just in reference to the blog post. Totally makes sense to add it in Test Butler too.

I don't think we'd need `executeShellCommand`. I think it would be [Settings.Secure#putInt](https://developer.android.com/reference/android/provider/Settings.Secure.html#putInt(android.content.ContentResolver, java.lang.String, int))

**Comment by bootstraponline** [comment]
Is there a startup script somewhere? That'd be cool to look at.

Thanks for the pointer to `Settings.Secure#putInt`

**Comment by drewhannay** [comment]
It's on the list of projects to open source, but not quite ready yet 😄
